duckduckgo/iOS
# enhance home page ux

Task/Issue URL: https://app.asana.com/0/392891325557410/788088018890007
Tech Design URL: 
CC:

**Description**:

Adds enhanced home screen functionality in the form of configurable favorites.

**Steps to test this PR**:

_iPad:_

1. Ensure that only a SERP variant gets assigned when on iPad
2. Ensure that previous static home screen is displayed on iPad

_Variant `mk` - control group_

1. Ensure that previous static home screen is displayed
1. Ensure that bookmarks works as expected

_Variant `ml` - single favorite_

1. Ensure that a single favorite is installed:
* Twitter - https://twitter.com/duckduckgo

_Variant `mm` - multiple favorites_

1. Ensure that three favorites are installed
* Twitter - https://twitter.com/duckduckgo
* Spread Privacy - https://spreadprivacy.com
* Quora - https://www.quora.com/Why-should-I-use-DuckDuckGo-instead-of-Google/answer/Gabriel-Weinberg

_Variants `ml` and `mm`_

1. Add / edit / delete favorites from the home screen
1. Reorder favorites from the home screen
1. Add / edit / remove bookmarks from a tab and the bookmarks screen
1. Reorder bookmarks in to favorites and favorites in to bookmarks from the bookmarks screen
1. Check landscape
1. Open, select and close multiple tabs
1. Check external launching behaves as expected (home long press buttons and widgets)
1. Check the following pixels are fired as appropriate:

* case homeScreenShown = "mh"
* case homeScreenSearchTapped = "mh_st"
* case homeScreenFavoriteLaunched = "mh_fl"
* case homeScreenAddFavorite = "mh_af"
* case homeScreenAddFavoriteOk = "mh_af_o"
* case homeScreenAddFavoriteCancel = "mh_af_c"
* case homeScreenEditFavorite = "mh_ef"
* case homeScreenDeleteFavorite = "mh_df"
* case homeScreenFavoriteMoved = "mh_fm"


---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
